src/context/registers.sw

```diff
@@ -65,8 +65,8 @@ pub fn context_gas() -> u64 {
     }
 }
 
-/// Received balance for this context.
-pub fn balance() -> u64 {
+/// Get the amount of units of `call_frames::msg_asset_id()` being sent.
+pub fn msg_amount() -> u64 {
```

Comment:
I would prefer to keep `balance` here because the fact that the balance register specifically is the message amount is an implementation detail. `msg_amount()` can use `balance()`